components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -1,110 +1,627 @@
+        // absorption optical depth, per layer [unitless]
+        auto odap_aer_icol = ekat::subview(aero_tau_lw, icol);
+        const auto atm     = mam_coupling::atmosphere_for_column(dry_atm, icol);
+
+        // FIXME: Get rid of this
+        // auto nccn = ekat::subview(aero_nccn, icol);
+
+        // FIXME: interface pressure [Pa]
+        auto pint = ekat::subview(p_int, icol);
+        // FIXME: dry mass pressure interval [Pa]
+        // FIXME:
+        auto zi      = ekat::subview(dry_atm.z_iface, icol);
+        auto pdel    = ekat::subview(p_del, icol);
+        auto pdeldry = ekat::subview(dry_atm.p_del, icol);
+
+        auto ssa_cmip6_sw_icol = ekat::subview(ssa_cmip6_sw, icol);
+        auto af_cmip6_sw_icol  = ekat::subview(af_cmip6_sw, icol);
+        auto ext_cmip6_sw_icol = ekat::subview(ext_cmip6_sw, icol);
+        auto ext_cmip6_lw_icol = ekat::subview(ext_cmip6_lw, icol);
+
+        // FIXME: check if this correct: Note that these variables have pver+1
+        // levels tau_w : aero_ssa_sw  (pcols,0:pver,nswbands) ! aerosol
+        // tau_w: aerosol single scattering albedo * tau
+        auto tau_w_icol = ekat::subview(aero_tau_ssa_sw, icol);
+        // tau_w_g:  (pcols,0:pver,nswbands)  aerosol assymetry
+        // parameter * tau * w
+        auto tau_w_g_icol = ekat::subview(aero_tau_g_sw, icol);
+        // tau_w_f(pcols,0:pver,nswbands) => aero_tau_forward  ?  aerosol
+        // forward scattered fraction * tau * w
+        auto tau_w_f_icol = ekat::subview(aero_tau_forward, icol);
+        // tau :  (pcols,0:pver,nswbands) aerosol
+        // aerosol extinction optical depth
+        auto tau_icol = ekat::subview(aero_tau_sw, icol);
+
+        auto work_icol = ekat::subview(work, icol);
+
+        // fetch column-specific subviews into aerosol prognostics
+        mam4::Prognostics progs =
+            mam_coupling::aerosols_for_column(dry_aero, icol);
+
+        mam4::aer_rad_props::aer_rad_props_sw(
+            team, dt, progs, atm, zi, pint, pdel, pdeldry, ssa_cmip6_sw_icol,
+            af_cmip6_sw_icol, ext_cmip6_sw_icol, tau_icol, tau_w_icol,
+            tau_w_g_icol, tau_w_f_icol, aerosol_optics_device_data, aodvis(icol), work_icol);
+
+        team.team_barrier();
+        mam4::aer_rad_props::aer_rad_props_lw(
+            team, dt, progs, atm, pint, zi, pdel, pdeldry, ext_cmip6_lw_icol,
+            aerosol_optics_device_data, odap_aer_icol);
+      });
+  Kokkos::fence();
+  // TODO: We will need to generate optical inputs files with  band ordering  that is consistent with
+  // RRTMGP
+  // Optical files depend on the band ordering in RRTMG. As temporal fix,
+  // we are correcting the band ordering of mam4xx's ouputs, so that they are consistent with
+  // inputs in RRTMG.
+  // Mapping from old RRTMG sw bands to new band ordering in RRTMGP
+  const int rrtmg_to_rrtmgp_swbands[14] = {13, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
```

Comment:
I don't think this static array is readable from device. You'll have to create a view. Probably best to do it an init, and keep the view as a class member.

Reply:
rrtmg_to_rrtmgp_swbands is now a class member and view.